src/executor.rs

```diff
@@ -0,0 +1,150 @@
+use std::mem::replace;
+use pyo3::PyResult;
+use revm::{Context, ContextWithHandlerCfg, Evm, EvmContext, FrameOrResult, FrameResult, inspector_handle_register};
+use revm::inspectors::TracerEip3155;
+use revm::primitives::{ExecutionResult};
+use revm::primitives::TransactTo;
+use revm_interpreter::{CallInputs, CreateInputs, SuccessOrHalt};
+use revm_interpreter::primitives::HandlerCfg;
+use crate::database::DB;
+use crate::utils::pyerr;
+
+/// Calls the EVM with the given context and handler configuration.
+pub(crate) fn call_evm(evm_context: EvmContext<DB>, handler_cfg: HandlerCfg, tracing: bool, is_static: bool) -> PyResult<(ExecutionResult, EvmContext<DB>)> {
+    if tracing {
+        let tracer = TracerEip3155::new(Box::new(crate::pystdout::PySysStdout {}), true);
+        let evm = Evm::builder()
+            .with_context_with_handler_cfg(ContextWithHandlerCfg {
+                cfg: handler_cfg,
+                context: Context {
+                    evm: evm_context,
+                    external: tracer,
+                },
+            })
+            .append_handler_register(inspector_handle_register)
+            .build();
+        run_evm(evm, is_static)
+    } else {
+        let evm = Evm::builder()
+            .with_context_with_handler_cfg(ContextWithHandlerCfg {
+                cfg: handler_cfg,
+                context: Context {
+                    evm: evm_context,
+                    external: (),
+                },
+            })
+            .build();
+        run_evm(evm, is_static)
+    }
+}
+
+/// Calls the given evm. This is originally a copy of revm::Evm::transact, but it calls our own output function
+fn run_evm<EXT>(mut evm: Evm<'_, EXT, DB>, is_static: bool) -> PyResult<(ExecutionResult, EvmContext<DB>)> {
```

Comment:
Didn't think that people were going to use handlers in this creative way! lol

Reply:
lol, creative often means there's a simpler way to do it? It's my first rust PR so please bear with me :laughing: